### PR TITLE
Add CRI-O support

### PR DESCRIPTION
**Pull request by McPatate**
I've committed a few files that shouldn't be there so that you guys can try out and test too. We should delete `test_k8s_crio` and `test.sh` if we ever want to merge this upstream.

The packer doesn't build yet there's just an error on the cleanup part

### CODE REVIEW

**Comment by christophe-rannou** [comment]
The AMI should also mount a NFS base on user data, or maybe you plan to do this in another PR ?

**Comment by McPatate** [comment]
> The AMI should also mount a NFS base on user data, or maybe you plan to do this in another PR ?

Yes I'll do another PR for this !

**Comment by McPatate** [comment]
X-referencing https://github.com/huggingface/spaces-app-manager/issues/83

File: scripts/install-worker.sh [hunk 1]
```diff
@@ -177,6 +177,72 @@ net.bridge.bridge-nf-call-iptables = 1
 net.ipv4.ip_forward = 1
 EOF
 
+###############################################################################
+### CRI-O setup ###############################################################
+###############################################################################
```
Comment (cbensimon) [inline scripts/install-worker.sh:182]: I think that the `kubelet-crio.service` file should be copied (from `$TEMPLATE_DIR` to `/etc/eks/crio`) in this section

File: files/bootstrap.sh [hunk 2]
```diff
@@ -479,6 +479,15 @@ if [[ "$CONTAINER_RUNTIME" = "containerd" ]]; then
     systemctl enable sandbox-image
     systemctl start sandbox-image
 
+elif [[ "$CONTAINER_RUNTIME" = "crio" ]]; then
+    sudo mkdir -p /etc/cni/net.d
+    sudo mv /etc/eks/containerd/kubelet-crio.service /etc/systemd/system/kubelet.service
```
Comment (cbensimon) [inline files/bootstrap.sh]:
```suggestion
    sudo mv /etc/eks/crio/kubelet-crio.service /etc/systemd/system/kubelet.service
```

**Comment by cbensimon** [comment]
Don't we need to add a `/etc/crio/crio.conf` file ? (from [here](https://medium.com/errnothxbye/centos-7-with-cri-o-on-eks-ae9684aff764#:~:text=o%20cri%2Dtools-,Configure%20CRI%2Do,-cat%20%3C%3C%20EOF%20%3E%20/etc))

File: scripts/install-worker.sh (hunk 1 above)
Comment (McPatate) [reply scripts/install-worker.sh:182]: yep that's correct, I haven't pushed my changes yet !

File: files/bootstrap.sh (hunk 2 above)
Comment (McPatate) [reply files/bootstrap.sh]: idem here

**Comment by McPatate** [comment]
> Don't we need to add a `/etc/crio/crio.conf` file ? (from [here](https://medium.com/errnothxbye/centos-7-with-cri-o-on-eks-ae9684aff764#:~:text=o%20cri%2Dtools-,Configure%20CRI%2Do,-cat%20%3C%3C%20EOF%20%3E%20/etc))

For now I think there's one generated by default when running `sudo make install.config`, but it may indeed be better to have that one copied.

**Comment by cbensimon** [comment]
> > Don't we need to add a `/etc/crio/crio.conf` file ? (from [here](https://medium.com/errnothxbye/centos-7-with-cri-o-on-eks-ae9684aff764#:~:text=o%20cri%2Dtools-,Configure%20CRI%2Do,-cat%20%3C%3C%20EOF%20%3E%20/etc))
> 
> For now I think there's one generated by default when running `sudo make install.config`, but it may indeed be better to have that one copied.

Ok,  so let's keep things simple and only use a custom config if needed

File: scripts/install-worker.sh [hunk 3]
```diff
@@ -177,6 +177,88 @@ net.bridge.bridge-nf-call-iptables = 1
...
+
+INSTALL_CRIO="${INSTALL_CRIO:-true}"
+if [[ "$INSTALL_CRIO" == "true" ]]; then
+    sudo mkdir -p /etc/eks/crio
+    if [ -f "/etc/eks/crio/crio.conf" ]; then
+        ## this means we are building a gpu ami and have already placed a containerd configuration file in /etc/eks
```
Comment (christophe-rannou) [inline scripts/install-worker.sh]:
```suggestion
        ## this means we are building a gpu ami and have already placed a crio configuration file in /etc/eks
```

**Review by christophe-rannou — APPROVED** [review]

**Pull request merged**